includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -344,6 +345,30 @@ public static function set_twentyseventeen_quotes_icon() {
 		} );
 	}
 
+	/**
+	 * Add filter to adjust the attachment image attributes to ensure attachment pages have a consistent <amp-img> rendering.
+	 *
+	 * This is only used in Twenty Seventeen.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/WordPress/wordpress-develop/blob/ddc8f803c6e99118998191fd2ea24124feb53659/src/wp-content/themes/twentyseventeen/functions.php#L545:L554
+	 */
+	public static function add_twentyseventeen_attachment_image_attributes() {
+		add_filter( 'wp_get_attachment_image_attributes', function ( $attr, $attachment, $size ) {
+			if ( ! is_attachment() ) {
+				return $attr;
+			}
+
+			$sizes = wp_get_attachment_image_sizes( $attachment->ID, $size );
+			if ( false !== $sizes ) {
+				$attr['layout'] = 'responsive';
```

Comment:
This is a a minor point, but would it be possible to not set the `layout` here, and let the image sanitizer [default to intrinsic](https://github.com/Automattic/amp-wp/blob/c236cc472ddcdfa22b373d4d32f24665034422f8/includes/sanitizers/class-amp-img-sanitizer.php#L247) (if there's a `height` and `width`)?

So far, it looks like the image displays the same with `intrinsic` and `responsive`. But you've tested this more 😄 

Of course, you have thought about this:
>Changing the layout to responsive ensures the image renders within the parent container in all viewports.

Reply:
@kienstra with the `intrinsic` layout, we then get into the issue over wider images overflowing the parent container.  But let me do a little more investigation to gather some test data for us.

Reply:
Thanks, @hellofromtonya!

Reply:
@kienstra Upon review, we can allow the layout to default and be handled in the image sanitizer.

Why?  The wide image overflow problem is tied to the Gutenberg image block.  That image block adjusts the width on the added parent `<figure>`.  When the layout is set to `intrinsic`, it sets the width to `fit-content`.  Combining that width with the inline width that AMP sets causes the overflowing problem.

Here we do not have that issue.  Therefore, we do not need to force the layout attribute. 

I've removed it in [commit 74dc97a](https://github.com/Automattic/amp-wp/pull/1321/commits/74dc97a0554eaed405c5fc26e7302248c628ff25).

Reply:
Thanks, @hellofromtonya! Sorry to nitpick 😄